From 4617c39ca212f0da82af1a9989c912bcdfe1158b Mon Sep 17 00:00:00 2001
From: Michael Biebl <biebl@debian.org>
Date: Sat, 3 Sep 2016 16:46:40 +0200
Subject: [PATCH] tests: use dbus-run-session instead of dbus-launch to run
 tests

The use of dbus-launch is deprecated by dbus upstream.
See https://lists.debian.org/debian-devel/2016/08/msg00554.html for more
details.

So use dbus-run-session instead, which was specifically designed to be
used for such cases, like running test-suites.

See also https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=836049.

https://bugzilla.gnome.org/show_bug.cgi?id=770798
---
 tests/Makefile.am | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index 233175c..3c907f1 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -135,7 +135,6 @@ EXTRA_DIST = \
 clean-local:
 	rm -f $(check_LTLIBRARIES:.la=.$(OS_EXT)) file.txt~
 
-DBUS_LAUNCH=$(shell which dbus-launch)
 RUN_TESTS_ENV_VARS= \
 	PYTHONPATH=$(top_builddir):$(top_builddir)/tests:$${PYTHONPATH:+:$$PYTHONPATH} \
 	LD_LIBRARY_PATH=$(builddir)/.libs:$$LD_LIBRARY_PATH \
@@ -149,10 +148,8 @@ RUN_TESTS_ENV_VARS= \
 # pygtkcompat tests need to be run in a separate process as they
 # clobber global name space
 check-local: $(check_LTLIBRARIES:.la=.$(OS_EXT)) $(test_typelibs) gschemas.compiled
-	export `$(DBUS_LAUNCH)` && \
-	$(RUN_TESTS_ENV_VARS) $(EXEC_NAME) $(PYTHON) -Wd $(srcdir)/runtests.py; rc=$$?; \
-	[ "$$rc" -ne 0 ] || [ -n "$$TEST_NAMES" ] || { TEST_NAMES=compat_test_pygtk $(RUN_TESTS_ENV_VARS) $(EXEC_NAME) $(PYTHON) -Wd -Werror::PendingDeprecationWarning -Werror::DeprecationWarning -Werror::RuntimeWarning $(srcdir)/runtests.py; rc=$$?; }; \
-	kill $$DBUS_SESSION_BUS_PID; \
+	$(RUN_TESTS_ENV_VARS) dbus-run-session -- $(EXEC_NAME) $(PYTHON) -Wd $(srcdir)/runtests.py; rc=$$?; \
+	[ "$$rc" -ne 0 ] || [ -n "$$TEST_NAMES" ] || { TEST_NAMES=compat_test_pygtk $(RUN_TESTS_ENV_VARS) dbus-run-session -- $(EXEC_NAME) $(PYTHON) -Wd -Werror::PendingDeprecationWarning -Werror::DeprecationWarning -Werror::RuntimeWarning $(srcdir)/runtests.py; rc=$$?; }; \
 	exit $$rc
 
 check.gdb:
-- 
2.9.3

